File: platform-paper/src/main/java/plutoproject/platform/paper/PlutoPaperLoader.java
package plutoproject.platform.paper;

import io.papermc.paper.plugin.loader.PluginClasspathBuilder;
import io.papermc.paper.plugin.loader.PluginLoader;
import plutoproject.framework.paper.dependency.DependenciesImpl;
import xyz.jpenilla.gremlin.runtime.platformsupport.PaperClasspathAppender;

import java.io.File;
import java.nio.file.Path;

@SuppressWarnings("unused")
public class PlutoPaperLoader implements PluginLoader {
    @Override
    public void classloader(PluginClasspathBuilder classpathBuilder) {
        Path cacheDirPath = classpathBuilder.getContext().getDataDirectory().resolve("libraries");
        File cacheDir = cacheDirPath.toFile();
        if (!cacheDir.exists()) {
            cacheDir.mkdirs();
        }
        new PaperClasspathAppender(classpathBuilder).append(DependenciesImpl.INSTANCE.resolve(cacheDirPath));
    }
}

File: platform-paper/src/main/kotlin/plutoproject/platform/paper/PlutoPaperLoader.kt
package plutoproject.platform.paper

import io.papermc.paper.plugin.loader.PluginClasspathBuilder
import io.papermc.paper.plugin.loader.PluginLoader
import plutoproject.framework.common.dependency.PaperDependencyResolver
import xyz.jpenilla.gremlin.runtime.platformsupport.PaperClasspathAppender

@Suppress("UNUSED")
class PlutoPaperLoader : PluginLoader {
    override fun classloader(classpathBuilder: PluginClasspathBuilder) {
        val cachePath = classpathBuilder.getContext().getDataDirectory().resolve("libraries")
        val cacheFile = cachePath.toFile()
        if (!cacheFile.exists()) {
            cacheFile.mkdirs()
        }
        PaperClasspathAppender(classpathBuilder).append(PaperDependencyResolver.resolve(cachePath))
    }
}